src/System.Device.Gpio/System/Device/Gpio/ExtendedPinMode.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Platform-dependent description of an alternate pin mode (i.e. a pin that is used for I2C or SPI transfer).
+    /// Depending on the hardware in use, the use will reflect the effective mode ("PWM", "SPI") or the state of the
+    /// mode multiplexer of the CPU ("ALT0", "ALT1").
+    /// </summary>
+    public class ExtendedPinMode : IEquatable<ExtendedPinMode>
+    {
+        private readonly string _modeName;
+
+        /// <summary>
+        /// The mode is not known
+        /// </summary>
+        public static readonly ExtendedPinMode Unknown = new ExtendedPinMode("Unknown");
+
+        /// <summary>
+        /// Creates a new mode.
+        /// </summary>
+        /// <param name="modeName">Common name of the mode</param>
+        protected ExtendedPinMode(string modeName)
+        {
+            _modeName = modeName;
+        }
+
+        /// <summary>
+        /// Equality implementation
+        /// </summary>
+        public virtual bool Equals(ExtendedPinMode? other)
+        {
+            if (ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            return other._modeName == _modeName;
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object? obj)
+        {
+            ExtendedPinMode? other = obj as ExtendedPinMode;
+            if (ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            return Equals(other);
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
```

Comment:
super minor but you may want to use the condense notation => for anything that just do a return or call a single function. Apply to this and the next 3 ones